test/tuple.jl

```diff
@@ -9,8 +9,59 @@ struct BitPerm_19352
     BitPerm_19352(xs::Vararg{Any,8}) = BitPerm(map(UInt8, xs))
 end
 
+# ntuples
+nttest1(x::NTuple{n, Int}) where {n} = n
+@test nttest1(()) == 0
+@test nttest1((1, 2)) == 2
+@test NTuple <: Tuple
+@test (NTuple{T, Int32} where T) <: Tuple{Vararg{Int32}}
+@test !((NTuple{T, Int32} where T) <: Tuple{Int32, Vararg{Int32}})
+@test Tuple{Vararg{Int32}} <: (NTuple{T, Int32} where T)
+@test Tuple{Int32, Vararg{Int32}} <: (NTuple{T, Int32} where T)
```

Comment:
I'm not entirely sure why these tests weren't there. I assume there's already some in there? I just moved this block over from core.jl